### PR TITLE
Header scope reduction (Airspeed and BattMonitor)

### PR DESCRIPTION
**Pull request by WickedShell**
I got sniped into chasing AP_Airspeed dependencies and didn't make it very far, but there was some easy wins for the AP_BattMonitor library, so a change in AP_BattMonitor.h now only triggers recompilation for files that directly use it (or live in the library). The mount dependency was silly, and the example was of no value since it didn't actually do anything.

### CODE REVIEW

**Comment by rmackay9** [comment]
Looks good to me.  I'm happy with the AP_Motors library change and with the removal of the examples.

**Comment by WickedShell** [comment]
Took this further, pulled `AP_GPS.h` out of `AHRS.h`, which moves it to only be in C++ files, vehicle headers, replay, and one scripting header that is only included within AP_Scripting source and doesn't escape.

File: libraries/AP_Airspeed/AP_Airspeed_Health.cpp [hunk 1]
```diff
@@ -30,8 +31,9 @@ void AP_Airspeed::check_sensor_ahrs_wind_max_failures(uint8_t i)
 
     // update state[i].failures.health_probability via LowPassFilter
     float speed_accuracy;
-    if (AP::gps().speed_accuracy(speed_accuracy)) {
-        const float gnd_speed = AP::gps().ground_speed();
+    const AP_GPS &gps = AP::gps();
+    if (gps.speed_accuracy(speed_accuracy)) {
+        const float gnd_speed = gps.ground_speed();
```
Comment (amilcarlucas) [inline libraries/AP_Airspeed/AP_Airspeed_Health.cpp]: This change does not realy improve readability, code size or speed. Or am I missing something?

Comment (WickedShell) [reply libraries/AP_Airspeed/AP_Airspeed_Health.cpp]: The intention was to not request the reference twice. I'm happy to back that part out to keep it at a minimal change set, but the general path we've taken has been to reuse the references rather then query for them.

File: libraries/AP_Airspeed/AP_Airspeed_Health.cpp [hunk 2]
```diff
@@ -2,6 +2,7 @@
 #include <GCS_MAVLink/GCS.h>
 #include <AP_Math/AP_Math.h>
 #include "AP_Airspeed.h"
+#include <AP_GPS/AP_GPS.h>
```
Comment (Jaaaky) [inline libraries/AP_Airspeed/AP_Airspeed_Health.cpp]: Please fix all windows-style end of line in this commit 157a3b1e34

**Pull request merged**